File: Data Structures & Algorithms/sum-of-two-integers/submission-1.java
class Solution {
    public int getSum(int a, int b) {
        int res = 0;
        int carry = 0;

        for (int i = 0; i < 32; i++) {
            int bitA = (a >> i) & 1;
            int bitB = (b >> i) & 1;
            int bitCur = bitA ^ bitB ^ carry;
            carry = bitA + bitB + carry >= 2 ? 1 : 0;

            if (bitCur == 1) {
                res = res | (1 << i);
            }    
        }
        
        return res;
    }
}

File: Data Structures & Algorithms/counting-bits/submission-1.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        if n <= 0:
            return [0]

        res = [0] * (n + 1)
        res[1] = 1
        for i in range(2, n + 1):
            if i % 2 == 0:
                res[i] = res[i // 2]
            else:
                res[i] = res[i - 1] + 1
        return res